File: android/app/src/main/java/app/ekhata/placeholder/PaymentLaunch.java
package app.ekhata.placeholder;

import android.net.Uri;

final class PaymentLaunch {
    private PaymentLaunch() {}

    static boolean isExternalPaymentUrl(Uri url) {
        if (url == null) {
            return false;
        }
        String scheme = url.getScheme();
        if (scheme == null) {
            return false;
        }
        switch (scheme.toLowerCase()) {
            case "upi":
            case "tez":
            case "gpay":
            case "phonepe":
            case "paytmmp":
            case "bhim":
            case "intent":
                return true;
            default:
                return false;
        }
    }
}

File: src/lib/paymentLaunch.ts
const PAYMENT_SCHEMES = new Set(['upi', 'tez', 'gpay', 'phonepe', 'paytmmp', 'bhim', 'intent'])

export function isExternalPaymentUrl(href: string): boolean {
  try {
    const url = new URL(href)
    return PAYMENT_SCHEMES.has(url.protocol.replace(':', '').toLowerCase())
  } catch {
    return /^(upi|tez|gpay|phonepe|paytmmp|bhim|intent):/i.test(href)
  }
}
